packages/react/src/components/List/ListContent/ListContent.jsx

```diff
@@ -157,17 +157,7 @@ const ListContent = ({
                 name={item.value}
                 data-testid={`${item.id}-checkbox`}
                 labelText=""
-                onClick={(evt) => {
-                  if (isSelectable && isCheckboxMultiSelect) {
-                    // When combing checkboxes with selectable rows (isSelectable) we are
-                    // getting click events from the ListItemWrapper before this handler is called.
-                    // Therefor we stop this one and rely on the ListItem onSelect handler to handle
-                    // the selection. If we don't the handleSelect callback will be called multiple times.
-                    evt.stopPropagation();
-                  } else {
-                    handleSelect(item.id, parentId);
-                  }
-                }}
```

Comment:
carbon seems to have fixed the double-event firing that was happening when clicking the checkbox, so this stopProp isn't needed anymore. You can confirm with: 

1. Go to story path=/story/1-watson-iot-list--with-checkbox-multi-selection
2. Verify that the multi selection works by clicking on the row, clicking the checkbox and using enter key when row is focused.
3. Verify the same but with knob isVirtualList checked
4. Go to story path=/story/1-watson-iot-list--with-checkbox-multi-selection-and-hierarchy
5. Repeat step 2-3 but also verify that child / parent selections works in BOTH directions including the indeterminate state
```